exercises/danny_hindson/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,89 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include <algorithm>
+#include <cctype>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+#include <string.h>
+#include <vector>
+
+bool compare(const std::pair<std::string,int>& x, const std::pair<std::string,int>& y) {
+  return x.second > y.second;
+}
+
+int main(int argc, char ** argv)
+{
+  std::string fileName;
+  if (argc == 2) {
+    fileName = argv[1];
+
+    std::string folderName ="D:\\newstarter\\builds\\WordCountDataFiles\\";
```

Comment:
We generally try to define constants used in a single file in an anonymous namespace in the file using uppercase naming conventions,

e.g. a string like this could be written in C++14 as
```
namespace {
    constexpr auto FOLDER_NAME = "D:\\newstarter\\builds\\WordCountDataFiles\\";
}
```
This would go just under the `include` lines and the anonymous namespace serves to keep it defined only in this file.